### ISSUE TITLE
Error messages (35)

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#337]
Previously

```
{
   "indexP"            : 0,
   ${indexP.${indexP}} : "newvalue"
}
```

caused

`'Could not set parameter '${indexP.${indexP}}' with value 'newvalue'! Reason: 'int' object is not subscriptable'`

And this is correct.

With latest changes in the JsonPreprocessor, the error message changed to:

`'expected string or bytes-like object'`

This seems to be an internal error and is not helpful for users. Please restore the previous error message.

**Comment by namsonx** [comment]
This ticket was solved on stabi branch.

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
merged for version 0.13.1
